src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -39,8 +42,50 @@
      * Андрей - 600
      * София - 900</p>
      */
+
+    private static final String DATE_PATTERN = "dd.MM.yyyy";
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
             throws Exception {
-        return null;
+
+        long dateFromInMillis = new SimpleDateFormat(DATE_PATTERN).parse(dateFrom).getTime();
+        long dateToInMillis = new SimpleDateFormat(DATE_PATTERN).parse(dateTo).getTime();
+
+        if (dateFromInMillis > dateToInMillis) {
+            throw new IllegalDateParametersException("Wrong parameters");
+        }
+
+        long[] arrayDatesInMillis = new long[data.length];
+
+        for (int i = 0; i < data.length; i++) {
+            arrayDatesInMillis[i] = new SimpleDateFormat(DATE_PATTERN)
+                    .parse(data[i].split(" ")[0]).getTime();
+        }
+
+        int[] salaryOfNames = new int[names.length];
```

Comment:
```suggestion
        int[] salariesByPersons = new int[names.length];
```
WDYT, is it variable name better and more suitable?